src/EnergyPlus/CurveManager.cc

```diff
@@ -3006,25 +3006,30 @@ namespace Curve {
         // Returns true if errors found
         Curve *thisCurve = state.dataCurveManager->PerfCurve(CurveIndex);
         int curveDim = thisCurve->numDims;
-        if (std::find(validDims.begin(), validDims.end(), curveDim) != validDims.end()) {
-            // Compatible
+        if (std::find(validDims.begin(), validDims.end(), curveDim) != validDims.end()) 
             return false;
-        } else {
-            // Not compatible
-            ShowSevereError(state, fmt::format("{}{}=\"{}\"", routineName, objectType, objectName));
-            ShowContinueError(state, format("...Invalid curve for {}.", curveFieldText));
-            std::string validString = fmt::to_string(validDims[0]);
-            for (std::size_t i = 1; i < validDims.size(); i++) {
-                validString += format(" or {}", validDims[i]);
-            }
-            std::string plural1 = curveDim > 1 ? "s" : "";
-            std::string plural2 = validDims[validDims.size() - 1] > 1 ? "s" : "";
-            ShowContinueError(state, format("...Input curve=\"{}\" has {} dimension{}.", thisCurve->Name, curveDim, plural1));
-            ShowContinueError(state, format("...Curve type must have {} dimension{}.", validString, plural2));
-            return true;
-        }
+
+        ErrorObjectHeader eoh{routineName, objectType, objectName};
+        std::string validString = fmt::to_string(validDims[0]);
+        for (std::size_t i = 1; i < validDims.size(); i++)  validString += format(" or {}", validDims[i]);
+
+        ShowErrorCurveDims(state, eoh, curveFieldText, thisCurve->Name, validString, curveDim);
+        return true;
     }
 
+    void ShowErrorCurveDims(EnergyPlusData &state,
+                            ErrorObjectHeader const &eoh,
+                            std::string_view fieldName,
+                            std::string_view curveName,
+                            std::string_view validDims,
+                            int dim)
+    {
+        ShowSevereError(state, fmt::format("{}{}=\"{}\"", eoh.routineName, eoh.objectType, eoh.objectName));
+        ShowContinueError(state, format("...Invalid curve for {}.", fieldName));
+        ShowContinueError(state, format("...Input curve=\"{}\" has dimension {}.", curveName, dim));
+        ShowContinueError(state, format("...Curve type must have dimension {}.", validDims));
+    }
+                           
```

Comment:
fieldName, curveName and validDims do not change in this function. Yes they are scalars but I want to know why not `const` on these. Did `dim` change?

Reply:
Adding `const` to "by value" variables (either local or parameter) doesn't do much to help the compiler. It's more a form of documentation for the programmer, and even then not a very meaningful one in my opinion. I'm just not in the habit of doing this. Old habits die hard, and while I am actively trying to break some old habits and develop new ones, I'm not sure this particular one is worth the cognitive effort.

Speaking of which, adding `const` to `&` or `*` variables definitely helps the compiler and is a meaningful form of documentation too so please continue doing that. That is also something I am not in the habit of doing, but in this case a habit I am actively trying to develop.